### DISCUSSION TITLE
The problem with reactor nerfing

### DISCUSSION THREAD

**Discussion by Notarin** [FakeFishGames/Barotrauma#10869] [Ideas and suggestions]
# Why nerfing the reactor isn't a good idea
I understand that the issue of overpowered two-component reactor controllers was a concern for the balance of the game, and I appreciate the efforts made to address it. However, after considering the implemented solution of slowing the movement of the sliders, I strongly believe that this approach is not the most effective or fair way to address the issue. I would like to respectfully present my reasoning and suggest alternative solutions for improving the balance of the game.
## Impact on game balance
One of the main problems with the slow movement of the sliders is that it significantly impacts the game's mechanics. By limiting the speed at which the sliders can be adjusted, you have effectively removed the possibility of creating more advanced controllers that could potentially perform better than the built-in auto mode. This not only removes a layer of challenge and creativity from the gameplay, but it also lowers the skill ceiling for building complicated devices. The slow movement of the sliders also does not make sense in the context of how other devices in the game function, as almost every other device follows values fed to it almost instantly. This inconsistency in the game's mechanics can be confusing and frustrating for players.
Despite the sliders being instant for humans to adjust, the intended manual control of the reactor remains unengaging and lacks complexity. This issue was present before the implementation of the current solution, which did not address the underlying problem. As a result, players have no motivation to manually control the reactor, leading to a less enjoyable gameplay experience. In order to address this issue, it would be beneficial to consider adding more depth and variety to the manual control system (The pressure control system was a wonderful addition), rather than simply slowing the movement of the sliders.
Another issue with the current solution is that it does not align with the overall design of the game. The game values player creativity and experimentation, and the slow movement of the sliders hinders these aspects of gameplay. Additionally, the slow movement of the sliders heavily contrasts with the instant response of other devices in the game to their inputs. By limiting the speed at which the sliders can be adjusted, the developers have effectively removed the possibility of creating more advanced controllers that could potentially perform better than the built-in auto mode. This not only limits player creativity and experimentation, but it also creates an inconsistency in the game's mechanics that can be confusing and frustrating for players.
## Proposing alternative solutions
It is evident that the current solution of slowing the movement of the sliders does not fully address the issues with game balance and design. While the developers may have had the best intentions in implementing this change, it is clear that a more comprehensive solution is needed to effectively address these problems. With that in mind, I would like to suggest some ideas that could potentially contribute to finding a more effective solution. These ideas are not meant to be the complete answer, but rather a starting point for further discussion and consideration. I believe that by working together, we can find a way to improve the game and provide a more enjoyable experience for all players.
 - ### Redesigning the control panel
	One possible solution to the issues with the current control system for the reactor would be to completely redesign the control panel. By focusing on creating an engaging and enjoyable experience for players, we can encourage them to spend more time manually controlling the reactor instead of relying on controllers. This approach would not only address the problems of the current system being too simple and uninteresting, but it could also help to balance the game by disincentivizing the use of overpowered controllers. By investing time and effort into improving the manual control of the reactor, we can create a more satisfying gameplay experience for players.
- ### Removing auto-calculated load
	Another potential solution to the issue of overpowered two-component reactor controllers could be to remove the auto-calculated load feature from the game. Currently, the reactor knows its requested load immediately, which makes it easy to build simple controllers that can outperform the built-in auto mode. By removing this functionality, players would have to manually calculate the load on the reactor by adding up the loads from the junction boxes. This would increase the complexity of both shipbuilding and manual reactor control, as players would need to keep an eye on the junction boxes or battery levels to see how the reactor's energy is being used. This would also create a new layer of challenge for players who want to build more advanced controllers, as they would need to perform more calculations to optimize the reactor's performance. Overall, this change could lead to a more engaging and satisfying gameplay experience.
- ### Overhauling reactor system
	Another possible solution to the issues with the current reactor system could be to completely revamp the way reactors work in the game. This could involve adding more variables and complexity to the reactors, such as taking into account the number of engines or other factors that affect their performance. Players would need to carefully adjust these variables in order to optimize the reactor's performance, and this would make it impossible to build a simple one-size-fits-all controller with just a few components. Instead, players would have to be aware of the specific build of the ship and account for it when creating a controller. This change could add a new layer of depth and challenge to the gameplay, and make building advanced controllers more engaging and rewarding.
- ### Slider movement damaging reactor
	One solution could be to implement a system where moving the sliders too quickly or excessively damages the reactor. One type of controller that is particularly problematic in this regard is the "bang-bang" controller, which works by rapidly moving the turbine slider back and forth between its minimum and maximum values. By causing the sliders to damage the reactor when moved too quickly, we can effectively remove the worst offenders and balance the game. This change would also encourage players to be more careful and precise when adjusting the sliders, adding a new layer of strategy and challenge to the gameplay.

I am eager to hear input and discuss these potential solutions further in order to find a solution that works for everyone. I understand that these are complex issues and that finding the right balance is important for the enjoyment of the game. I am open to hearing other ideas and suggestions, and I hope that we can work together to find a way to improve the game and create a more satisfying experience for all players. I am committed to finding a solution that addresses the underlying issues and benefits the game as a whole.
The current solution to the problem of overpowered two-component reactor controllers is inadequate and has negative impacts on gameplay, game balance, and game design. While the developers may have had good intentions in implementing this change, it is clear that a more comprehensive and thoughtful solution is needed. By considering alternative approaches such as redesigning the control panel, removing the auto-calculated load feature, overhauling the reactor system, or causing slider movement to damage the reactor, we can find a way to address the underlying issues and create a more enjoyable and balanced gameplay experience. I hope that we can work together to find a solution that benefits the game and all of its players.

I would like to add that I apologise if this is the wrong location to post this, or that I somehow missed a post that has already discussed this. I know that there are some posts that discuss similar subjects, but I felt that my post was different enough to warrant its own space, sorry if I stepped on any toes, I really didn't mean to.

**Comment by BiffW** [comment]
Agreed. Limiting reactor's responding speed is more like a ban to auto reactor than a fair solution. The slider movement based damage sounds like a fair solution to me.

**Comment by kleiner-sentry** [comment]
Perhaps the reactor was destined to always be automated. It's like trying to make manually operating a coal power plant fun. It'll probably never be fun in the first place and trying to make it fun would be like banging your head on the wall trying to think of good ideas.

I take another route to this solution. I think we ought to embrace automating the reactor completely, in favor of using barotraumas talent system to the Engineers advantage. We should keep the way the reactor functions as it is right now, minus to slider moving penalty, and some pins for the pressure gauges if we really wanted to. we'll then give the engineer special talents that allows them special buffs, and the ability to redirect power to specific systems on the sub, overvolting the selected systems, and undervolting everything else. This would be the best way to encourage team communication and engagement, and thus solve the problem.

Another way to involve the Engineer in the going ons of the reactor is to integrate these reactor controllers into the autopilot of the reactor itself. We can do this by creating these chips, that then automatically control the reactor when the autopilot is flicked on. We can even add quality to these controllers, adding additional buffs to the reactor, encouraging the engineer to seek out these special chips, or craft it themselves to provide the crew with a better reactor.

**Comment by MateusJK15** [comment]
I don't understand why they nerfed a feature that is completely optional, default subs don't have ARCs, the player need to manually create a sub to add it, nerfing it not only made the game worse but also failed at making the player more interested in manually controlling the reactor, it's still a very boring task (by lack of complexity) that i would just assign a bot to do it.

**Reply by Notarin** [reply]
I can definitely see where you're coming from regarding the nerfing of the ARCs. However, I do believe that there was a legitimate balancing issue with the way the reactors were set up before. It was too easy for new players to create an effective reactor without fully understanding the features, and without putting in the effort to actually explore and experiment with the controls.

That being said, I think that the solution implemented by the developers was not acceptable thus I'm proposing alternative solutions. I strongly believe that a complete overhaul of the reactor system is necessary to address these issues and make the experience of controlling the reactor more enjoyable and engaging.

**Comment by Biluga2** [comment]
I think excessive slider movement damaging the reactor is a good idea, to prevent bang bang systems working whille precise systems still work... To me it's just sad bcs i have a lot of sub ideas but i cant make any of them if i cant make a reactor that doesnt kill every junction box, currentely i was trying to solve the Overvoltage problem with batteries recharge rate, when my auto reactor control cant keep up because of the nerf but im not being able make it work right really frustrating for me... This is just sad/annoying

**Reply by Regalis11** [reply]
I know this doesn't solve the underlying issue, but have you considered using some kind of "dummy load" (for example a supercapacitor) to smooth out the overvoltage? I feel there are still plenty of ways to create very efficient automatized power systems, it just takes a bit more creativity and effort nowadays when you can't just have a reactor that responds to voltage changes near-instantaneously.

**Reply by VaelophisNyx** [reply]
still sad that this response didn't address that no one wants to do reactor work atm due to it being painful and boring at the same time. ARCs were a reaction to unengaging gameplay and the fix wasn't banning them, it was fixing the gameplay

**Comment by pavel4win** [comment]
reactor/battery automation complex solution for prevent cable box damage
On Russian language (use google trans)
This is my first version. Now I developed more universal and simple solution. Someday I will write it.

https://steamcommunity.com/sharedfiles/filedetails/?id=3246995944
